manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -11,8 +11,8 @@ metadata:
     {{- .Values.annotations | toYaml | nindent 4 }}
 spec:
   {{- if not .Values.autoscaling.enabled }}
-  {{- with .Values.replicaCount }}
-  replicas: {{ . }}
+  {{- if and (hasKey .Values "replicaCount") (ne .Values.replicaCount nil) }}
```

Comment:
A simpler `replicas: {{ .Values.replicaCount }}` (without the `if` condition) would work for me as well, but I'm trying to preserve the previous behavior as closely as possible in case there are valid reasons for it...

I guess preventing an invalid `replicas: ` field with an empty value seems like a good reason to keep the `if` condition. Otherwise, we risk generating invalid k8s YAML if `replicaCount` isn't set.